### ISSUE TITLE
Problem with datetime format

### ISSUE THREAD

**Issue by RomanBuzuk** [go-swagger/go-swagger#1084]
## Problem statement

 I tried to send request to get server time, but i get zero time. I have similar code, that works with usual string, and this code works well, but with strfmt.datetime I have problem . As I can see, server send payload with right time, but client receive zero time. Can you help me to find problem? 
Or maybe you can give example, how to work with strfmt.datetime.

My code:
Server:
configure_api.go:
```
api.GetTimeHandler = operations.GetTimeHandlerFunc(func(params operations.GetTimeParams) middleware.Responder {
		t := models.Datetime(time.Now())
		fmt.Printf("Server time %v \n", time.Time(t).String())
		return operations.NewGetTimeOK().WithPayload(t)
	})
```
Client:
```
type Client struct {
	http *apiclient.ConfgenSwagger
}
func (c *Client) GetTime() (models.Datetime, error) {
	p := operations.NewGetTimeParams()
	resp, err := c.http.Operations.GetTime(p)
        if err != nil {
		fmt.Printf("Client time error  \n")
		return resp.Payload, err
	}
	t := resp.Payload
	fmt.Printf("Client time %v \n", time.Time(t).String())
	return t, err
}
```

I get next results:
```
Server time 2017-06-29 17:07:26.956884177 +0300 MSK
Client time 0001-01-01 00:00:00 +0000 UTC
```
## Swagger specification

```
...
datetime:
    type: string
    format: date-time
paths:
  /time:
    get:
      description: get server time
      operationId: getTime
      responses:
        '200':
          description: getTime success
          schema:
            $ref: "#/definitions/datetime"
        default:
          description : getTime error
          schema:
            $ref: "#/definitions/error"
```

## Environment
swagger version: 0.9.0  
go version: 1.5.3

**Comment by RomanBuzuk** [comment]
It seems like that client and server never use functions MarshalJSON and UnmarshalJSON from go-openapi/strfmt/time.go.

**Comment by RomanBuzuk** [comment]
I think, I found the reason of bug.
When we call ProducerFunc from /go-openapi/runtime/json.go ,then  data interface{} has type Model.Datetime ( type, that is generated by goswagger). But this type has not methods like MarshalJSON. So it can not make right JSON.
But Model.Datetime is just redefinition of type strfmt.DateTime, that has such methods. 
So we have two possible ways of solution of problem:
1)Add automatic generation of methods like MarshalJSON for generated types 
```
func (t Datetime) MarshalJSON() ([]byte, error) {
	return strfmt.DateTime(t).MarshalJSON()
}
```
2) Convert type Model.Datetime to strfmt.DateTime before it falls into ProducerFunc

**Comment by casualjim** [comment]
it implementes the TextMarshaller which the json marshalling should make use of.

**Comment by fredbi** [comment]
I stepped on that too when handling default or enum values.
The type aliasing cuts any call to the unmarshalling methods made available by the strfmt type.

I believe we should generate the MarshallJSON method for all situations where the type is aliased to a strfmt type which is not a string alias (currently these are: Date, DateTime, Duration, ObjectID).

**Comment by fredbi** [comment]
ok I started a small PR on this which fixes this annoyance. test is ongoing.

@casualjim do you think we should add the [Un]MarshalBinary interface as well for these simple type redefinitions? (I could never catch what was the use case for these)

**Comment by fredbi** [comment]
Fixed with #1420
